deepmd/train/trainer.py

```diff
@@ -412,14 +442,12 @@ def _init_session(self):
                 log.info('receive global variables from task#0')
             run_sess(self.sess, bcast_op)
 
-    def train (self, train_data, valid_data=None) :
+    def train (self, train_data = None, valid_data=None) :
 
         # if valid_data is None:  # no validation set specified.
         #     valid_data = train_data  # using training set as validation set.
 
         stop_batch = self.stop_batch
-        self._init_session()
-
```

Comment:
Why moving `_init_session` to `_build_training`?

Reply:
We need self.saver within the _init_session to store the checkpoint, in the old implementation we must call trainer.train to init the self.save

Reply:
It doesn't make sense to init session in build training. 

one may call it in _do_works.

Reply:
I'll move it back